auth/src/main/scala/mesosphere/marathon/example/plugin/auth/ExampleAuthenticator.scala

```diff
@@ -39,18 +45,48 @@ class ExampleAuthenticator extends Authenticator with PluginConfiguration {
       }
     }
 
+    loadIdentities()
+
     for {
       auth <- request.header("Authorization").headOption
       (username, password) <- basicAuth(auth)
-      identity <- identities.get(username) if BCrypt.checkpw(password, identity.password)
+      identity <- identities.get(username) if checkPassword(password, identity)
     } yield identity
 
   }
 
+  private def checkPassword(password: String, identity: ExampleIdentity): Boolean = {
+    try {
+      BCrypt.checkpw(password, identity.password)
+    } catch {
+      case e:Exception =>
+        log.error(s"Error while checking password", e)
```

Comment:
This seems unrelated to the work being done. Was it picked up during testing? What did the uncaught exception do?

Reply:
The caught exception resulted in 500 responses without any explanation on what is going wrong. I ran into it, and changed it for the better.